### ISSUE TITLE
Bind mounting problem, second run of an app fails

### ISSUE THREAD

**Issue by keleset** [flatpak/flatpak#3851]
## Linux distribution and version
Alpine Edge
## Flatpak version
1.8.1
## Description of the problem
So, the problem is: Steam installs and runs successfully, but only once. Any subsequent execution ends instantly with:
```
flatpak run -v com.valvesoftware.Steam
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/me/.local/share/flatpak
F: Opening user flatpak installation at path /home/me/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/me/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Cleaning up unused container id 2982173205
F: Allocated instance id 823989937
F: Add defaults in dir /com/valvesoftware/Steam/
F: Add locks in dir /com/valvesoftware/Steam/
F: Allowing wayland access
F: Allowing x11 access
F: Allowing pulseaudio access
F: Pulseaudio user configuration file '/home/me/.config/pulse/client.conf': Error opening file /home/me/.config/pulse/client.conf: No such file or directory
F: Failed to run in transient scope: No systemd user session available, cgroups not available
F: Running 'bwrap --args 33 xdg-dbus-proxy --args=35'
F: Running 'bwrap --args 33 steam-wrapper'
bwrap: Can't bind mount /bindfileBHIoNa on /newroot/home/me/.var/app/com.valvesoftware.Steam/config/user-dirs.dirs: Invalid argument
```
Full `-vv` log here: `https://pastebin.com/1GMnDbUz`

## Steps to reproduce
1) Install Alpine Edge
2) Install Flatpak
3) Install Steam from Flathub
4) Run Steam, play some games, have fun
5) Shutdown steam, have some rest
6) No more fun until Steam flatpak is reinstalled with `--delete-data`

Separate issue from https://github.com/flatpak/flatpak/issues/3807 because of different mount points being affected.
Actually only Steam flatpak affected by the issue, but there was a recommendation to open an issue here (https://github.com/flathub/com.valvesoftware.Steam/issues/602).

**Comment by PureTryOut** [comment]
I didn't have this problem before, but I just reinstalled my PC this weekend and can now reproduce this. I'm also using Alpine Linux so that might be a cause somehow, but then again it worked fine before.
